### DISCUSSION TITLE
Command line based debugger for C# running on .NET 6

### DISCUSSION THREAD

**Discussion by awa5114** [dotnet/csharplang#6203] [Q&A]
[Crossposting from SE](https://stackoverflow.com/questions/72548721/debug-c-sharp-from-the-command-line)

I'm on a (possibly futile) quest to find a command line based debugger for C# applications running on the .NET 6 framework. Something similar to `jdb` for java or `pdb` for python. I'm on a windows machine. Here are the meagre resources I have found:

- [debugging c# in the command line](https://stackoverflow.com/questions/1656864/debugging-c-sharp-in-the-command-line)
- [MDbg](https://stackoverflow.com/questions/2791459/where-can-i-get-mdbg)
- [DebugExe flag for `devenv`](https://docs.microsoft.com/en-us/visualstudio/ide/reference/debugexe-devenv-exe?view=vs-2022)

The most promising by far would have been `MDbg`, but this looks very much outdated. [The page on the NuGet repository][1] shows no version compatible with .NET 6.

I also made a [post][2] on reddit  but this unfortunately did not generate much discussion or answers.

Being able to hop into the debugger from the command line is a common feature among several languages, so I feel like it should exist for C#, however convoluted the process may be.

Is there any way to do this in C#?

  [1]: https://www.nuget.org/packages/MDbg/
  [2]: https://www.reddit.com/r/csharp/comments/uny9jn/command_line_based_debugger_for_csharp/

**Comment by CyrusNajmabadi** [comment]
@awa5114 This is the repo for the C# language design.  I recommend asking these sorts of questions in Discord or StackOverflow, etc.  Thanks!

**Comment by HaloFour** [comment]
At best this might be a question for the [runtime](https://github.com/dotnet/runtime) as to what tooling they support for debugging.  The C# language doesn't know about debugging and the compiler emits debug symbols but doesn't get involved in the tools expected to consume it.

**Reply by awa5114** [reply]
Fair enough, I'll ask it there then...
